### PR TITLE
Use conditional compile

### PR DESCRIPTION
**Pull request by markcowl**
Failures in TypeSpecValidation are a known bug in TSV, where it validates more folders than were actually changed

### CODE REVIEW

File: eng/pipelines/templates/steps/typespec-ci.yml [hunk 1]
```diff
@@ -14,7 +14,14 @@ steps:
     pushd ${{parameters.Folder}}
 
     # Pass "--no" to ensure npx does not install anything (TypeSpec compiler should already be installed)
-    npx --no tsp compile . --warn-as-error || exit_code=1
+    if test -f tspconfig.yaml; then
```
Comment (mikeharder) [inline eng/pipelines/templates/steps/typespec-ci.yml:17]:
```suggestion
```

You can remove this check.  We should only call `templates/steps/typespec-ci.yml` on folders already known to contain `tspconfig.yaml`.

File: eng/pipelines/templates/steps/typespec-ci.yml [hunk 2]
```diff
@@ -14,7 +14,14 @@ steps:
...
+      if test -f main.tsp; then
+        npx --no tsp compile . --warn-as-error || exit_code=1
+      fi
+      if test -f main.tsp; then
+        npx --no tsp compile client.tsp --no-emit --warn-as-error || exit_code=1
+      fi
+    fi
```
Comment (mikeharder) [inline eng/pipelines/templates/steps/typespec-ci.yml:24]:
```suggestion
```

Comment (mikeharder) [inline eng/pipelines/templates/steps/typespec-ci.yml:24]:
```suggestion
```

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: eng/pipelines/templates/steps/typespec-ci.yml (hunk 2 above)
Comment (markcowl) [reply eng/pipelines/templates/steps/typespec-ci.yml:20]: I think i resolved this, lmk if not let me know

File: specification/cognitiveservices/HealthInsights/healthinsights.oncophenotype/tspconfig.yaml [hunk 3]
```diff
@@ -1,25 +1,19 @@
...
+    clear-output-folder: true
+    new-project: false
+    model-namespace: false
+  "@azure-tools/cadl-python":
+    package-mode: "dataplane"
+    package-name: "azure-healthinsights-cancerprofiling"
+  "@azure-tools/cadl-typescript": 
+    packageDetails:
+      name: "@azure-rest/health-insights-cancerprofiling"
+      description: "Azure Health Insights CancerProfiling"
```
Comment (mikeharder) [inline specification/cognitiveservices/HealthInsights/healthinsights.oncophenotype/tspconfig.yaml:19]:
```suggestion
      description: "Azure Health Insights CancerProfiling"

```

**Review by mikeharder — APPROVED** [review]
LGTM, but please wait for review from @timotheeguerin

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**